### PR TITLE
Merge Sign up page and BE authentication for signup

### PR DESCRIPTION
**Pull request by thuymymai**

### CODE REVIEW

File: server/controllers/authController.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const passport = require('passport');
+const { httpError } = require('../utils/error');
+
+const login = (req, res, next) => {
+  passport.authenticate('local', { session: false }, (err, user, info) => {
+    console.log('user passed from cb', user);
+    if (err || !user) {
+      return res.status(400).json({
+        message: 'Incorrect email of password. Enter again.',
+        user: user
```
Comment (GiangNguyen1207) [inline server/controllers/authController.js:12]: Why you want to send the user to front end when there is no user?

File: server/controllers/authController.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      };
+      const token = jwt.sign(user, process.env.JWT_SECRET);
+      return res.json({ user, token });
+    });
+  })(req, res, next);
+};
+
+const register = (req, res, next) => {
+  passport.authenticate('local', { session: false }, (err, user, info) => {
+    console.log('user passed from signup', user);
```
Comment (GiangNguyen1207) [inline server/controllers/authController.js]: remove the console.log here

File: server/controllers/authController.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    if (err || !user) {
+      return res.status(400).json({
+        message: 'Enter again.',
```
Comment (GiangNguyen1207) [inline server/controllers/authController.js]: can the message be more contentful?

File: server/controllers/authController.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        user: user
```
Comment (GiangNguyen1207) [inline server/controllers/authController.js]: the same for log in

File: server/controllers/userController.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+const user_get = async (req, res) => {
+	const user_retrieved = await userModel.getUser(req.params.id);
+	console.log('user_get_id', req.params.id)
+	res.json(user_retrieved);
+};
+
+const user_create_post = (req, res, next) => {
+	const errors = validationResult(req);
+	if (!errors.isEmpty()) {
+		console.log("error:", errors.array());
```
Comment (GiangNguyen1207) [inline server/controllers/userController.js:21]: remove console log

File: server/controllers/userController.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+		const err = httpError('Data not valid', 400);
+		next(err);
+	} else {
+		const user_created = userModel.addUser(req.body);
+		console.log(req.body);
```
Comment (GiangNguyen1207) [inline server/controllers/userController.js:26]: here as well

File: server/utils/passport.js [hunk 7]
```diff
@@ -1 +1,81 @@
...
+  usernameField: 'email',
+  passwordField: 'password',
+  passReqToCallback: true // allows us to pass back the entire request to the callback
+},
+  async (req, email, password, done) => {
+    try {
+      const existUser = await getUserByEmail(email);
+      const firstName = req.body.firstname;
+      const lastName = req.body.lastname;
+      console.log("name in sign up", firstName, lastName);
```
Comment (GiangNguyen1207) [inline server/utils/passport.js]: try to remove all the console log in every file

**Review by Nischhal3 — APPROVED** [review]

**Pull request closed without merging**